src/argilla/server/apis/v0/handlers/users.py

```diff
@@ -52,7 +56,13 @@ async def whoami(request: Request, current_user: models.User = Security(auth.get
 
     await telemetry.track_login(request, username=current_user.username)
 
-    return User.from_orm(current_user)
+    user = User.from_orm(current_user)
+
+    if current_user.is_admin:
+        workspaces = accounts.list_workspaces(db)
+        user.workspaces.extend([workspace.name for workspace in workspaces])
```

Comment:
Correct me if I'm wrong but I believe you can finish with duplications in the workspaces. For example when the user has one workspace you are adding that same workspace at the end.

Reply:
If you fix this please add a test for this scenario.